src/Symfony/Bridge/Doctrine/Messenger/EventSubscriber/EntityMessageCollector.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Messenger\EventSubscriber;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Doctrine\ORM\Events;
+use Symfony\Bridge\Doctrine\Messenger\EntityMessage\EntityMessageCollectionInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\DispatchAfterCurrentBusStamp;
+
+/**
+ * Doctrine listener that listens to Persist, Update and Remove. Every time this is
+ * invoked we take messages from the entities.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+class EntityMessageCollector implements EventSubscriber
+{
+    private $messageBus;
+
+    public function __construct(MessageBusInterface $messageBus)
+    {
+        $this->messageBus = $messageBus;
+    }
+
+    public function getSubscribedEvents()
+    {
+        return [
+            Events::postFlush,
+        ];
+    }
+
+    public function postFlush(LifecycleEventArgs $event)
```

Comment:
This code will fail.
The `postFlush` event is not a lifecycle event. It passes an instance of `Doctrine\ORM\Event\PostFlushEventArgs` which is not a subclass of `LifecycleEventArgs`. See https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/events.html#postflush .

Reply:
So this event cannot be used here at all, because at this point you do not know which entities were flushed.